packages/yew-macro/src/html_tree/lint/mod.rs

```diff
@@ -0,0 +1,102 @@
+//! Lints to catch possible misuse of the `html!` macro use. At the moment these are mostly focused
+//! on accessibility.
+
+use proc_macro_error::emit_warning;
+use syn::spanned::Spanned;
+
+use crate::props::{ElementProps, Prop};
+
+use super::{html_element::HtmlElement, HtmlTree};
+
+/// Lints HTML elements to check if they are well formed. If the element is not well-formed, then
+/// use `proc-macro-error` (and the `emit_warning!` macro) to produce a warning. At present, these
+/// are only emitted on nightly.
+pub trait Lint {
+    fn lint(element: &HtmlElement);
+}
+
+/// Applies all the lints to the HTML tree.
+pub fn lint_all(tree: &HtmlTree) {
+    lint::<AHrefLint>(tree);
+    lint::<ImgAltLint>(tree);
+}
```

Comment:
Maybe make this a macro?
```rust
lint_all!(AHrefLint, ImgAltLint)
```
seems better than the current implementation

Reply:
Where would i be getting `tree` then?

Reply:
why?

Reply:
> Where would i be getting `tree` then?

Ah, right, I overlooked that

> Why would it be better?

I initially said that because we would've been able to combine all `lint` calls in one line. Feel free to ignore this comment. It doesn't really matter